pylint/checkers/typecheck.py

```diff
@@ -1198,24 +1174,24 @@ def _get_nomember_msgid_hint(
         node: nodes.Attribute | nodes.AssignAttr | nodes.DelAttr,
         owner: SuccessfulInferenceResult,
     ) -> tuple[Literal["c-extension-no-member", "no-member"], str]:
-        suggestions_are_possible = self._suggestion_mode and isinstance(
-            owner, nodes.Module
+        if _is_c_extension(owner):
+            return "c-extension-no-member", ""
+        if not self.linter.config.missing_member_hint:
+            return "no-member", ""
+        names = _similar_names(
+            owner,
+            node.attrname,
+            self.linter.config.missing_member_hint_distance,
+            self.linter.config.missing_member_max_choices,
         )
-        if suggestions_are_possible and _is_c_extension(owner):
-            msg = "c-extension-no-member"
-            hint = ""
+        if not names:
+            return "no-member", ""
+        names = [repr(name) for name in names]
+        if len(names) == 1:
+            names_hint = ", ".join(names)
```

Comment:
I know it was this way before, but how about:
```suggestion
            names_hint = names[0]
```